target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/platform.sh

```diff
@@ -69,8 +71,100 @@ platform_check_image() {
 	return 0;
 }
 
+glinet_validate_firmware() {
+	local type=$(identify_magic_long $(get_magic_long $1))
+	if [ "$type" = "unknown 73797375" ]; then type="sys"; fi
+
+	echo "$type"
+}
+
+glinet_do_fit_upgrade() {
+	echo -n "fit: Extract [ FIT IMAGE ] -x-x-> [ ubi.bin ] ... "
+	local ubi=/tmp/ubi.bin
+	local part=$(dumpimage -l /tmp/firmware.bin | grep -o "Image [0-9] (ubi)" | cut -f2 -d" ")
+
+	local ubibin=$( dumpimage -T flat_dt -p ${part} -o "$ubi"  $1 )
```

Comment:
No need to store the output of the command in an unused variable. However, there should be an error management (even if it is implicitly done below with the `-s` test, I think it is always clearer to catch the error as soon as possible):

```suggestion
	if ! dumpimage -T flat_dt -p "$part" -o "$ubi" "$1"; then
		echo "Failed to extract UBI image"
		# error management
	fi
```